examples/with-docker/yarn.lock

```diff
@@ -0,0 +1,1604 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+"@babel/code-frame@7.12.11":
```

Comment:
Could we remove this file? 🙏 We don't commit the lock files since the next version is usually `latest`.

Reply:
Hello @leerob , thanks for the comments. I have removed the `yarn.lock` which meant changing the dockerfile too.

Reply:
Sorry, I wasn't clear enough - the lockfile should be part of `gitignore`, not removed from the Docker file 👍

Reply:
If the lockfile is not in the codebase (git ignored) and in the `Dockerfile`, the docker build will fail, probably I am not understanding correctly.

Next.js has a [yarn.lock](https://github.com/vercel/next.js/blob/canary/yarn.lock) and the one on the root is not [ignored](https://github.com/vercel/next.js/blob/canary/.gitignore#L9) -- others [are](https://github.com/vercel/next.js/blob/canary/.gitignore#L8).

Open to suggestions :)

Reply:
I'm talking specifically about the .`gitignore` for the `with-docker` example, which this PR edits: https://github.com/vercel/next.js/blob/canary/examples/with-docker/.gitignore

We do _not_ want to commit lockfiles here, since the version of `next` is `latest` in the `package.json`: https://github.com/vercel/next.js/blob/canary/examples/with-docker/package.json#L10

If the Dockerfile is generating a lock file, let's add it to that ignore file so it's not committed to the repo. Otherwise, when someone clones this example (e.g. `yarn create next-app --example with-docker nextjs-docker`) then it would already have a lock file with an out of date version of Next.

Reply:
Done: https://github.com/vercel/next.js/pull/25824/files#diff-8e402cdfd8a1759fbbcb4776892c795205bea98ef959c59ebc69bbc16f68ef84R36 , thanks!